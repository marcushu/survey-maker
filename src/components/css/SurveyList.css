table {
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 50px;
} 
th {
    color: rgb(1, 85, 134);
    padding: 15px;
}


td {
    margin: 5px;
    padding: 15px;
    border-bottom: solid;
    border-color: rgb(1, 85, 134);
    border-width: 1px;
    background-color: rgb(215, 221, 226);
}

td.respondents {
    text-align: center;
    font-weight: bold;
}

.graphicButtons {
    border: none;
    background-color: rgb(233 233 233);
}