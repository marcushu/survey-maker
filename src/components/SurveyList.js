import React from 'react';
import people from '../images/people-fill.svg';
import envelope from '../images/envelope-fill.svg'
import PropTypes from 'prop-types';
import './css/SurveyList.css';

export default function SurveyList({
  surveyTypes,
  goToStats,
  addParticipants,
  createUrls,
  deleteSurvey }) {


  return (
    <table style={{ width: "100%" }}>
      <thead>
        <tr>
          <th>Survey (view/edit)</th>
          <th style={{width: "75px"}}>Respondents</th>
        </tr>
      </thead>
      <tbody>
        {surveyTypes
          ? surveyTypes.map(surveyItem =>
            <tr key={surveyItem._key}> 
              <td>
                <button className="surveyBtn"
                  onClick={() => goToStats(surveyItem._key)}>
                  {surveyItem._key}
                </button>
              </td>
              <td className="respondents">
                {surveyItem._val - 1}
              </td>
              <td style={{  width: "135px" }}>
                <button
                  title="Add survey participants"
                  className="graphicButtons"
                  onClick={() => addParticipants(surveyItem._key)}>
                  <img src={people} alt="new participants"></img>
                </button>
                  &nbsp;
                <button
                  title="Create email URL's"
                  className="graphicButtons"
                  onClick={() => createUrls(surveyItem._key)}>
                  <img src={envelope} alt="email urls"></img>
                </button>
                  &nbsp;
                <button
                  onClick={() => deleteSurvey(surveyItem._key)}
                  className="graphicButtons">
                    X
                </button>
              </td>
            </tr>
          )
          : <tr>
            <td colSpan="2"> <h5 style={{ color: "orange" }}>Loading...</h5></td>
          </tr>
        }
      </tbody>
    </table>
  )
}

SurveyList.propTypes = {
  surveyTypes: PropTypes.array,
  goToStats: PropTypes.func,
  addParticipants: PropTypes.func,
  createUrls: PropTypes.func,
  deleteSurvey: PropTypes.func
}