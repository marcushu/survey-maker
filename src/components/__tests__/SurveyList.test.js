import React from 'react';
import {  render, fireEvent } from '@testing-library/react';
import SurveyList from '../SurveyList';

describe('SurveyList', () => {
    const mockFunc = jest.fn();

    const mockTypesArray = [
        {_key: "Gallery Show", _val: 5},
        {_key: "Valentines Receptions", _val: 1}
    ]

    it("renders", () => {
        const { asFragment } = render(
            
            <SurveyList
            surveyTypes={mockTypesArray}
            goToStats={mockFunc}
            addParticipants={mockFunc}
            createUrls={mockFunc}
            deleteSurvey={mockFunc} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("correctly renders survey name and participant count", () => {
        const { getByRole } = render(
            <SurveyList
            surveyTypes={mockTypesArray}
            goToStats={mockFunc}
            addParticipants={mockFunc}
            createUrls={mockFunc}
            deleteSurvey={mockFunc} />
        );

        expect(getByRole('cell', { name: /4/i })).toBeInTheDocument();
        expect(getByRole('cell', { name: /0/i })).toBeInTheDocument();
        expect(getByRole('cell', { name: /gallery show/i })).toBeInTheDocument();
        expect(getByRole('cell', { name: /valentines receptions/i })).toBeInTheDocument();
    
    });

    it("sends the correct information to callbacks", () => {
        const { getAllByTitle } = render(
            <SurveyList
            surveyTypes={mockTypesArray}
            goToStats={mockFunc}
            addParticipants={mockFunc}
            createUrls={mockFunc}
            deleteSurvey={mockFunc} />
        );
 
        fireEvent.click(getAllByTitle("Add survey participants")[0]);

        expect(mockFunc).toHaveBeenCalledTimes(1);
        expect(mockFunc.mock.calls[0][0]).toBe("Gallery Show");

    })
})