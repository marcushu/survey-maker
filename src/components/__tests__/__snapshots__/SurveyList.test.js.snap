// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyList renders 1`] = `
<DocumentFragment>
  <table
    style="width: 100%;"
  >
    <thead>
      <tr>
        <th>
          Survey (view/edit)
        </th>
        <th
          style="width: 75px;"
        >
          Respondents
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button
            class="surveyBtn"
          >
            Gallery Show
          </button>
        </td>
        <td
          class="respondents"
        >
          4
        </td>
        <td
          style="width: 135px;"
        >
          <button
            class="graphicButtons"
            title="Add survey participants"
          >
            <img
              alt="new participants"
              src="people-fill.svg"
            />
          </button>
           
          <button
            class="graphicButtons"
            title="Create email URL's"
          >
            <img
              alt="email urls"
              src="envelope-fill.svg"
            />
          </button>
           
          <button
            class="graphicButtons"
          >
            X
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            class="surveyBtn"
          >
            Valentines Receptions
          </button>
        </td>
        <td
          class="respondents"
        >
          0
        </td>
        <td
          style="width: 135px;"
        >
          <button
            class="graphicButtons"
            title="Add survey participants"
          >
            <img
              alt="new participants"
              src="people-fill.svg"
            />
          </button>
           
          <button
            class="graphicButtons"
            title="Create email URL's"
          >
            <img
              alt="email urls"
              src="envelope-fill.svg"
            />
          </button>
           
          <button
            class="graphicButtons"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
